### PR TITLE
Auv simulation

### PR DESCRIPTION
**Pull request by NRottmann**
Adding files which enable a simulation with the autonomous underwater
vehicle (AUV) from the Technical University Hamburg-Harburg.

For safety reasons I just added files (only had to add some lines to the CMakeFile in sitl_gazebo and to the config file posix_sitl_default). I have tested the simulation with gazebo-ROS and it works fine.

### CODE REVIEW

**Comment by TSC21** [comment]
@NRottmann CI is failing. Can you check and fix it up?

**Comment by NRottmann** [comment]
@TSC21 I checked it. It seems that something is wrong with my mixer, but I am not sure what's wrong. Can you have a look into it?

**Comment by TSC21** [comment]
There must be something on your mixer that doesn't pass the mixer test - https://github.com/PX4/Firmware/blob/a6c682ce504a33f64a3bde83e5f9004c252a476a/src/systemcmds/tests/test_mixer.cpp

**Comment by TSC21** [comment]
Still problems with the mixer...

**Comment by TSC21** [comment]
Try putting the same mixer in `ROMFS/px4fmu_test/mixers` also

**Comment by TSC21** [comment]
@NRottmann did you add the model sample/uaage to thw Devguide already?

**Comment by NRottmann** [comment]
@TSC21 no I did not. Shall I? How to do that?

File: ROMFS/px4fmu_test/mixers/uuv_quad_x.mix [hunk 1]
```diff
@@ -1,38 +1,21 @@
-MuM AUV Mixer for HippoCampus
-============================
-
-This file defines mixers for MuM HippoCampus suitable for testing.
-Scaling down to 100% of full throttle.
-
-CCW 1
-----
```
Comment (TSC21) [inline ROMFS/px4fmu_test/mixers/uuv_quad_x.mix]: I don't think that removing section titles would prejudice the CI.

File: posix-configs/SITL/init/lpe/hippocampus [hunk 2]
```diff
@@ -13,7 +13,7 @@ param set CAL_MAG0_ID 196616
 simulator start -s
 
 pwm_out_sim mode_pwm
-mixer load /dev/pwm_output0 ROMFS/px4fmu_common/mixers/uuv_quad_x.main.mix
+mixer load /dev/pwm_output0 ROMFS/px4fmu_test/mixers/uuv_quad_x.mix
```
Comment (TSC21) [inline posix-configs/SITL/init/lpe/hippocampus]: Does this really make a difference? Supposedly px4fmu_test mixers are for tests only, but you have a real vehicle mixer though.

**Comment by TSC21** [comment]
> @TSC21 no I did not. Shall I? How to do that?

https://github.com/PX4/Devguide

**Comment by TSC21** [comment]
@NRottmann can you update?

**Comment by LorenzMeier** [comment]
New rebased PR with cleaned commit: https://github.com/PX4/Firmware/pull/7828

**Pull request closed without merging**